gtfs-realtime-validator-lib/src/main/java/edu/usf/cutr/gtfsrtvalidator/lib/validation/rules/FrequencyTypeZeroValidator.java

```diff
@@ -119,6 +139,126 @@
+
+        if (!inTripUpdates(currentTripUpdate, previousTripUpdates)) {
+            if (!isNewTrip(previousTripUpdates, currentTripUpdate)) {
+                String errorMessage = "vehicle_id " + currentTripUpdate.getVehicle().getId() + " startTime has changed and is now " + currentTripUpdate.getTrip().getStartTime();
+                RuleUtils.addOccurrence(E053, errorMessage, errorListE053, _log);
+            }
+        }
+
+    }
+
+    private boolean isNewTrip(HashMap<TripUpdateKey, List<TripUpdate>> previousTripUpdates, TripUpdate tripUpdate) {
+        TripUpdate latest = null;
+        List<TripUpdate> tripUpdates = previousTripUpdates.get(new TripUpdateKey(tripUpdate.getVehicle().getId(), tripUpdate.getTrip().getTripId()));
+        if (tripUpdates != null) {
+            for (TripUpdate previousTripUpdate : tripUpdates) {
+                if (latest == null) {
+                    latest = previousTripUpdate;
+                }
+                if (new TripUpdateStartTimeComparator().compare(latest, previousTripUpdate) > 0) {
+                    latest = previousTripUpdate;
+                }
+            }
+            if (new TripUpdateStartTimeComparator().compare(tripUpdate, latest) > 0) {
+                return true;
+            } else {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private boolean inTripUpdates(TripUpdate tripUpdate, HashMap<TripUpdateKey, List<TripUpdate>> previousTripUpdates ) {
+
+        List<TripUpdate> tripUpdates = previousTripUpdates.get(new TripUpdateKey(tripUpdate.getVehicle().getId(), tripUpdate.getTrip().getTripId()));
+        if(tripUpdates !=null )
+        {
+            for (TripUpdate previousTripUpdate : tripUpdates) {
+                if (previousTripUpdate.getTrip().getStartTime().equals(tripUpdate.getTrip().getStartTime())) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+
+    private void addTripUpdate(HashMap<TripUpdateKey, List<TripUpdate>> tripUpdatesMap, GtfsRealtime.TripUpdate tripUpdate) {
+        if (tripUpdate.hasVehicle() && tripUpdate.getVehicle().hasId()) {
+
+            List<TripUpdate> tripUpdates = null;
+            if (!tripUpdatesMap.containsKey(new TripUpdateKey(tripUpdate.getVehicle().getId(), tripUpdate.getTrip().getTripId()))) {
+                tripUpdates = new ArrayList<TripUpdate>();
+            } else {
+                tripUpdates = tripUpdatesMap.get(new TripUpdateKey(tripUpdate.getVehicle().getId(), tripUpdate.getTrip().getTripId()));
+            }
+            tripUpdates.add(tripUpdate);
+
+            //Collections.sort(tripUpdates, new TripUpdateStartTimeComparator());
+
+            tripUpdatesMap.put(new TripUpdateKey(tripUpdate.getVehicle().getId(), tripUpdate.getTrip().getTripId()), tripUpdates);
+        }
+    }
+    private class TripUpdateKey
```

Comment:
I think you could just concatenate the vehicle_id and trip_id strings and that by itself would be fine for a composite key?  So "1234_400" or something similar.  Having an entire class for this seems overblown to me.